packages/CollapsibleCard/src/CollapsibleCard.js

```diff
@@ -1,83 +1,71 @@
-  const [metadataId] = React.useState(() => `collapsible-card-metadata_${uuidv4()}`);
-  const {
-    "CollapsibleCard.Avatar": avatar,
-    "CollapsibleCard.Metadata": metadata,
-    children: otherElements,
-  } = extractChildren(children, ["CollapsibleCard.Avatar", "CollapsibleCard.Metadata"]);
-
-  React.useEffect(() => {
-    if (!isCollapsed && onExpand) {
-      onExpand();
-    }
-  }, [isCollapsed, onExpand]);
+  const { children, initialIsCollapsed, isEditing, isFirstRow, isLastRow, isMiddleRow, onToggleIsCollapsed } = props;
+  const [isCollapsed, setIsCollapsed] = React.useState(initialIsCollapsed);
 
-  function getLabel() {
-    return (
-      <sc.Label>
-        {avatar}
-        <div>
-          <sc.LabelText id={labelTextId}>{label}</sc.LabelText>
-          {metadata ? React.cloneElement(metadata, { id: metadataId }) : null}
-        </div>
-      </sc.Label>
-    );
+  function handleToggleIsCollapsed() {
+    onToggleIsCollapsed(!isCollapsed);
+    setIsCollapsed(oldIsCollapsed => !oldIsCollapsed);
   }
 
-  function handleClick() {
-    setIsCollapsed(prevIsCollapsed => !prevIsCollapsed);
-  }
+  const thingsToShare = {
+    handleToggleIsCollapsed,
+    isCollapsed,
+    isEditing,
+    isFirstRow,
+    isLastRow,
+    isMiddleRow,
+    onToggleIsCollapsed,
+  };
 
   return (
-    <sc.CollapsibleCard
-      iconAlign="right"
-      label={typeof label === "function" ? label({ isCollapsed }) : getLabel()}
-      isCollapsed={isCollapsed}
-      onClick={handleClick}
-      hasAvatar={!!avatar}
-      hasLabelOnly={!avatar && !metadata}
-      triggerAriaDescribedby={`${labelTextId} ${metadata ? metadataId : ""}`}
-      {...moreProps}
-    >
-      <sc.Content hasDivider={hasDivider}>{otherElements}</sc.Content>
-    </sc.CollapsibleCard>
+    <CollapsibleCardContext.Provider value={thingsToShare}>
+      <sc.CollapsibleCard
+        aria-expanded={!isCollapsed}
+        isCollapsed={isCollapsed}
+        isEditing={isEditing}
+        isFirstRow={isFirstRow}
+        isMiddleRow={isMiddleRow}
+        isLastRow={isLastRow}
+      >
+        {children}
+      </sc.CollapsibleCard>
+    </CollapsibleCardContext.Provider>
   );
 }
 
 const propTypes = {
   children: PropTypes.node,
-
-  /** If has a divider between collapsible header and content. */
-  hasDivider: PropTypes.bool,
-
-  /** Label text as the card title, can be a render function. */
-  label: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
-
-  /** Callback function when expand the card. */
-  onExpand: PropTypes.func,
+  initialIsCollapsed: PropTypes.bool,
+  isEditing: PropTypes.bool,
+  isFirstRow: PropTypes.bool,
+  isMiddleRow: PropTypes.bool,
+  isLastRow: PropTypes.bool,
+  onToggleIsCollapsed: PropTypes.func,
 };
 
 const defaultProps = {
   children: null,
-  hasDivider: false,
-  label: null,
-  onExpand: null,
+  initialIsCollapsed: true,
+  isEditing: false,
+  isFirstRow: null,
+  isMiddleRow: null,
+  isLastRow: null,
```

Comment:
I don't leave un-necessary props around! 😄 

isEditing gives it the yellow background.
isFirstRow/isMiddleRow/isLastRow are used to decide where the card needs to have rounded corners, if any.  `position` or `index` wouldn't differentiate between `isMiddleRow` and `isLastRow`.

Reply:
I mean the 3 props `isFirstRow ` `isMiddleRow` `isLastRow` is for interal use inside `CollapsibleCard` component. Will the consumer use them? if not why not just combine them together, to make the list of props shorter

Reply:
I think even better would be to solve this with pure CSS i.e. `:first-child`/`:last-child`/`:not(:first-child):not(:last-child)`
if not possible, then I agree having all three of these as props is redundant. It's always better to avoid the potential for invalid combinations of props (even though it's all internal in this case). E.g. have a `position` of `"first"`/`"middle"`/`"last"` instead.

Reply:
Oh i see what you mean. Ok I can use one `position` string prop instead of three booleans.  I dont think pure css will work.